src/driver/xen.rs

```diff
@@ -164,6 +184,86 @@ impl Introspectable for Xen {
         }))
     }
 
+    fn listen(&mut self, timeout: u32) -> Result<Option<Event>, Box<dyn Error>> {
+        let fd = self.xev.xenevtchn_fd()?;
+        let fd_struct = PollFd::new(fd, PollFlags::POLLIN | PollFlags::POLLERR);
+        let mut fds = [fd_struct];
+        let mut vcpu: u16 = 0;
+        let mut event_type = unsafe { mem::MaybeUninit::<EventType>::zeroed().assume_init() };
+        let poll_result = poll(&mut fds, timeout.try_into().unwrap()).unwrap();
```

Comment:
use a `match` on the poll result to handle each possible value, and `_` for all the others.